server/routerlicious/packages/routerlicious-base/src/riddler/mongoTenantRepository.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ICollection } from "@fluidframework/server-services-core";
+import { ITenantDocument } from "./tenantManager";
+
+/**
+ * Abstract away ITenant collection logic
+ */
+export interface ITenantRepository {
+	/**
+	 * Finds one query in the database
+	 *
+	 * @param query - data we want to find
+	 * @param options - optional. If set, provide customized options to the implementations
+	 * @returns The value of the query in the database.
+	 */
+	findOne(query: any, options?: any): Promise<ITenantDocument>;
+
+	/**
+	 * @returns All values in the database.
+	 */
+	findAll(): Promise<ITenantDocument[]>;
```

Comment:
We should not expose this kind of api, I think we have to add some limit. `findAll` is not realistic, I think we should add some limit.

Given current `tenants` collection is unsharded, all queries will hit one partition. This means it will hit the RU limit 10K in one partition and will cause server side to retry and add latency. Depends on our luck, we may experience throttling. Even the `tenantsV2` becomes sharded, `findAll` is not a good api to use for the same partition limit, networkIO purpose. 

Could we look where we use this `findAll` usage in both OSS and FRS and see if we could change to with pagination instead?

Reply:
Resolved findAll() calls by referring to the FRS PR https://msazure.visualstudio.com/One/_git/FRS/pullrequest/7515135